devtools/parcel/main.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	texttemplate "text/template"
+
+	"golang.org/x/net/html"
+)
+
+var tpl = `package webparcel
+
+func init() {
+	ParcelAssetMap = {{ $.AssetMap }}
+}
+`
+
+func readHtmlFromFile(fileName string) (string, error) {
+	bs, err := ioutil.ReadFile(fileName)
+
+	if err != nil {
+		return "", err
+	}
+
+	return string(bs), nil
+}
+
+func ParseNameWithHash(hashedName string) string {
+	extension := path.Ext(hashedName)
+
+	nameWithHash := strings.TrimSuffix(hashedName, extension)
+	dotIdx := strings.LastIndex(nameWithHash, ".")
+
+	nameOnly := nameWithHash[:dotIdx]
+
+	return fmt.Sprintf("%s%s", nameOnly, extension)
+}
+
+func isModuleScript(a []html.Attribute) bool {
+	for _, v := range a {
+		if v.Key == "type" && v.Val == "module" {
+			return true
+		}
+	}
+	return false
+}
+
+func isStylesheet(a []html.Attribute) bool {
+	for _, v := range a {
+		if v.Key == "rel" && v.Val == "stylesheet" {
+			return true
+		}
+	}
+	return false
+}
+
+func getAssetHashedName(a []html.Attribute, key string) string {
+	for _, v := range a {
+		if v.Key == key {
+			return strings.TrimPrefix(v.Val, "/")
+		}
+	}
+	return ""
+}
+
+func getAssetMap(doc *html.Node) map[string]string {
+	var m = make(map[string]string)
+
+	var traverse func(n *html.Node) *html.Node
+	traverse = func(n *html.Node) *html.Node {
+		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			if c.Data == "script" {
+				hashedName := getAssetHashedName(c.Attr, "src")
+				if isModuleScript(c.Attr) {
+					m[fmt.Sprintf("%s-module", ParseNameWithHash(hashedName))] = hashedName
+				} else {
+					m[ParseNameWithHash(hashedName)] = hashedName
+				}
+			}
+			if c.Data == "link" && isStylesheet(c.Attr) {
+				hashedName := getAssetHashedName(c.Attr, "href")
+				m[ParseNameWithHash(hashedName)] = hashedName
+			}
+
+			res := traverse(c)
+			if res != nil {
+				return res
+			}
+		}
+		return nil
+	}
+	traverse(doc)
+
+	return m
+}
+
+func main() {
+	htmlText, err := readHtmlFromFile("./resources/authgear/static/build.html")
```

Comment:
The convention of a well-behaved command line program is to exit with 1 in case of error.

```typescript
func main() {
  err := do()
  if err != nil {
    fmt.Fprintf(os.Stderr, "%v\n", err)
    os.Exit(1)
  }
}
```